docs/resources/alert.md

```diff
@@ -0,0 +1,162 @@
+
+* `id` - unique ID of the Alert.
+* `lifecycle_state` - The workspace state of the alert. Used for tracking trashed status. (Possible values are `ACTIVE` or `TRASHED`).
+* `state` - Current state of the alert's trigger status (`UNKNOWN`, `OK`, `TRIGGERED`). This field is set to `UNKNOWN` if the alert has not yet been evaluated or ran into an error during the last evaluation.
+* `create_time` - The timestamp string indicating when the alert was created.
+* `update_time` - The timestamp string indicating when the alert was updated.
+* `trigger_time` - The timestamp string when the alert was last triggered if the alert has been triggered before.
+
+## Migrating from `databricks_sql_alert` resource
+
+Under the hood, the new resource uses the same data as the `databricks_sql_alert`, but exposed via different API. This means that we can migrate existing alerts without recreating them.  This operation is done in few steps:
+
+* Record the ID of existing `databricks_sql_alert`, for example, by executing the `terraform state show databricks_sql_alert.alert` command.
+* Create the code for the new implementation performing following changes:
+  * the `name` attribute is now named `display_name`
+  * the `parent` (if exists) is renamed to `parent_path` attribute, and should be converted from `folders/object_id` to the actual path.
+  * `options` block is converted into `condition` block with the following changes:
+    * the value of `op` attribute should be converted from mathematical operator into string name, like, `>` is becoming `GREATER_THAN`, `==` is becoming `EQUAL`, etc.
+    * the `column` attribute is becoming `operand` block
+    * the `value` attribute is becoming `threshold` block.  **Please note that old implementation always used strings so you may have changes after import if you use `double_value` or `bool_value` inside the block.**
+  * the `rearm` attribute is renamed to `seconds_to_retrigger`.
+  
+For example, if we have the original `databricks_sql_alert` defined as:
+
+```hcl
+resource "databricks_sql_alert" "alert" {
+  query_id = databricks_sql_query.this.id
+  name     = "My Alert"
+  parent   = "folders/${databricks_directory.shared_dir.object_id}"
+  options {
+    column = "value"
+    op     = ">"
+    value  = "42"
+    muted  = false
+  }
+}
+```
+
+we'll have a new resource defined as:
+
+```hcl
+resource "databricks_alert" "alert" {
+  query_id     = databricks_sql_query.this.id
+  display_name = "My Alert"
+  parent_path  = databricks_directory.shared_dir.path
+  condition {
+    op = "GREATER_THAN"
+    operand {
+      column {
+        name = "value"
+      }
+    }
+    threshold {
+      value {
+        double_value = 42
+      }
+    }
+  }
+}
+```
+
+* Remove the old resource from the state with the `terraform state rm databricks_sql_alert.alert` command.
+* Import new resource with the `terraform import databricks_alert.alert <alert-id>` command.
```

Comment:
ok - let me look into it. We still may need to have migration guide for TF < 1.7.x because some people still on the old versions due to the re-licensing